src/klein/_app.py

```diff
@@ -39,17 +44,30 @@
 from ._decorators import modified, named
 from ._interfaces import IKleinRequest
 from ._resource import KleinResource
-from ._typing import Awaitable, KwArg, VarArg
 
 
 KleinSynchronousRenderable = Union[str, bytes, IResource, IRenderable]
 KleinRenderable = Union[
     KleinSynchronousRenderable, Awaitable[KleinSynchronousRenderable]
 ]
-KleinRoute = Callable[[Any, IRequest, VarArg(Any), KwArg(Any)], KleinRenderable]
-KleinErrorHandler = Callable[
-    [Optional["Klein"], IRequest, Failure], KleinRenderable
-]
+
+
+class KleinRoute(Protocol):
+    def __call__(self, request: IRequest) -> KleinRenderable:
```

Comment:
Note that this removes `*args` and `**kwargs`.

If those a present, then any method decorated with `@route` has to accept them, and that's not currently (and doesn't want to be) a requirement.